File: appengine/monorail/framework/logger.py
# Copyright 2022 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.
""""Helper methods for structured logging."""

from __future__ import print_function
from __future__ import division
from __future__ import absolute_import

import google.cloud.logging

import settings


def log(struct):
  if settings.local_mode or settings.unit_test_mode:
    return

  logging_client = google.cloud.logging.Client()
  logger = logging_client.logger('python')
  logger.log_struct(struct)

File: go/src/infra/cros/recovery/internal/execs/logger.go
// Copyright 2022 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package execs

import (
	"context"

	"infra/cros/recovery/logger"
	"infra/cros/recovery/tlw"
)

// NewLogger returns logger.
func (ei *ExecInfo) NewLogger() logger.Logger {
	return ei.runArgs.Logger
}

// GetLogRoot returns path to logs directory.
func (ei *ExecInfo) GetLogRoot() string {
	return ei.runArgs.LogRoot
}

// CopyFrom copies files from resource to localhost.
func (ei *ExecInfo) CopyFrom(ctx context.Context, resourceName, srcFile, destDir string) error {
	return ei.runArgs.Access.CopyFileFrom(ctx, &tlw.CopyRequest{
		Resource:        resourceName,
		PathSource:      srcFile,
		PathDestination: destDir,
	})
}

// CopyDirectoryFrom copies a directory from resource to localhost.
func (ei *ExecInfo) CopyDirectoryFrom(ctx context.Context, resourceName, srcDir, destDir string) error {
	return ei.runArgs.Access.CopyDirectoryFrom(ctx, &tlw.CopyRequest{
		Resource:        resourceName,
		PathSource:      srcDir,
		PathDestination: destDir,
	})
}
